### PR TITLE
Update to React 16 and Enzyme 3

### PR DESCRIPTION
**Pull request by gregaubert**
Main updates in the tests due to Enzyme 3 changes : 
* Add `wrapper.update()` when changing state or props from outside or asynchronously
* `componentDidMount` is now called when shallow rendering : 
  * Disable it in some cases with `disableLifecycleMethods`
  * Remove useless code in tests that was mimicking the `componentDidMount`
  * Fix some mocked WS that needed to return Promise
* Update snapshots with : 
  * `undefined` properties that are now displayed
  * empty string `""` instead of `null`
* Mock createPortals for modals testing because it's not well supported yet in current version of Enzyme

### CODE REVIEW

File: server/sonar-web/src/main/js/store/utils/configureStore.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ import thunk from 'redux-thunk';
 const middlewares = [thunk];
 const composed = [];
 
-if (process.env.NODE_ENV !== 'production') {
+if (!['production', 'test'].includes(process.env.NODE_ENV)) {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/store/utils/configureStore.js]: why?

File: server/sonar-web/src/main/js/apps/quality-profiles/components/ProfileContainer.tsx [hunk 2]
```diff
@@ -32,11 +33,10 @@ interface Props {
   onRequestFail: (reasong: any) => void;
   organization: string | null;
   profiles: Profile[];
-  router: { replace: (path: any) => void };
   updateProfiles: () => Promise<void>;
 }
 
-export default class ProfileContainer extends React.PureComponent<Props> {
+export default class ProfileContainer extends React.PureComponent<Props & WithRouterProps> {
```
[the remark this answers is not in the document]
Comment (gregaubert) [reply server/sonar-web/src/main/js/apps/quality-profiles/components/ProfileContainer.tsx]: Yes but I didn't really find a nice way for testing...

File: server/sonar-web/src/main/js/store/utils/configureStore.js (hunk 1 above)
Comment (gregaubert) [reply server/sonar-web/src/main/js/store/utils/configureStore.js]: I'll switch to `=== 'development'`

File: server/sonar-web/src/main/js/apps/organizations/components/forms/RemoveMemberForm.js [hunk 3]
```diff
@@ -60,6 +60,12 @@ export default class RemoveMemberForm extends React.PureComponent {
     this.closeForm();
   };
 
+  handleClose = (e /*: MouseEvent */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/organizations/components/forms/RemoveMemberForm.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/apps/organizations/components/forms/ManageMemberGroupsForm.js [hunk 4]
```diff
@@ -104,6 +104,12 @@ export default class ManageMemberGroupsForm extends React.PureComponent {
     this.closeForm();
   };
 
+  handleClose = (e /*: MouseEvent */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/organizations/components/forms/ManageMemberGroupsForm.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/apps/organizations/components/forms/AddMemberForm.js [hunk 5]
```diff
@@ -72,6 +72,12 @@ export default class AddMemberForm extends React.PureComponent {
     }
   };
 
+  handleClose = (e /*: MouseEvent */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/organizations/components/forms/AddMemberForm.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-web/src/main/js/apps/organizations/components/forms/AddMemberForm.js [hunk 6]
```diff
@@ -72,15 +72,22 @@ export default class AddMemberForm extends React.PureComponent {
...
+    e.preventDefault();
+    e.stopPropagation();
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/organizations/components/forms/AddMemberForm.js]: remove this

File: server/sonar-web/src/main/js/apps/organizations/components/forms/ManageMemberGroupsForm.js [hunk 7]
```diff
@@ -104,11 +104,18 @@ export default class ManageMemberGroupsForm extends React.PureComponent {
...
+    e.preventDefault();
+    e.stopPropagation();
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/organizations/components/forms/ManageMemberGroupsForm.js]: remove this

File: server/sonar-web/src/main/js/apps/organizations/components/forms/RemoveMemberForm.js [hunk 8]
```diff
@@ -60,11 +60,18 @@ export default class RemoveMemberForm extends React.PureComponent {
...
+    e.preventDefault();
+    e.stopPropagation();
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/organizations/components/forms/RemoveMemberForm.js]: remove this

File: server/sonar-web/src/main/js/apps/organizations/components/forms/__tests__/__snapshots__/AddMemberForm-test.js.snap [hunk 9]
```diff
@@ -2,72 +2,23 @@
...
+    key="add-member-button"
+    onClick={[Function]}
+  />,
   <Modal
-    contentLabel="modal form"
+    contentLabel="users.add"
+    key="add-member-modal"
     onRequestClose={[Function]}
-  >
-    <header
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/organizations/components/forms/__tests__/__snapshots__/AddMemberForm-test.js.snap]: 🤔

Comment (stas-vilchik) [reply server/sonar-web/src/main/js/apps/organizations/components/forms/__tests__/__snapshots__/AddMemberForm-test.js.snap]: try with jest 21.3.0-beta.4

**Pull request merged**